### DISCUSSION TITLE
Dataset bias ("❤️ Translated by Amara.org Community")

### DISCUSSION THREAD

**Discussion by ExtReMLapin** [openai/whisper#928] [Q&A] [answered]
Hello,

I noticed multiples biases using whisper.

For example, it sometimes outputs (in french) `❤️ Translated by Amara.org Community` as I guess it was used video subtitles by Amara.

There are also leftovers of "soustitreur.com" which implies OpenAI used soustitreur.com as a contractor.

And at the end of a music video, when music fades away, it outputed : "Thank you for watching".

Have you guys noticed any bias like this ?

**Comment by ExtReMLapin** [comment]
related to #651

**Reply by EtienneAb3d** [reply]
See also #679

**Comment by glangford** [comment]
I saw this today in a transcription from Portuguese. It was at the end of a video (newscast) with no voice present and closing theme music playing:

`Legendas pela comunidade Amara.org`

**Reply by ExtReMLapin** [reply]
Here is a list of (mostly french) lines i got 👍🏻


```python

filter_out_biases_statics = {
    " Merci d'avoir regardé cette vidéo.",
    " Merci d'avoir regardé cette vidéo!",
    " Merci d'avoir regardé la vidéo.",
    " J'espère que vous avez apprécié la vidéo.",
    " Je vous remercie de vous abonner",
    " Sous-titres réalisés para la communauté d'Amara.org",
    " Sous-titres réalisés para la communauté d'Amara.org"
    " Merci d'avoir regardé!",
    " ❤️ par SousTitreur.com",
    " — Sous-titrage ST'501 —"
    " Thanks for watching!",
}

```

**Reply by ryanheise** [reply]
Building a central database of these sounds like a good idea -- maybe in a python file in this repo?

**Reply by ExtReMLapin** [reply]
Here is what I did to get these lines.
As we have hallucinations with silences and background noises, I decided to add with pydub a splitter that strips silences and splits the audio in multiple files, more files, more biases like this as it shows up at the end.

But the issue is there are ~100 languages to deal with, so the list is going to be LONG.

**Reply by ghost** [reply]
I got `Transcribed by https://otter.ai` today. After transcribing the final sentence, there is 2 seconds of silence before ending when that got added. At first I was worried I had a virus. Why is this happening?

I used https://github.com/Const-me/Whisper , model large (v2) https://huggingface.co/ggerganov/whisper.cpp/tree/main https://github.com/ggerganov/whisper.cpp

**Comment by doublex** [comment, marked as answer]
This is our list:
````json
{
  "en": [
    " www.mooji.org",
  ],
  "nl": [
    " Ondertitels ingediend door de Amara.org gemeenschap",
    " Ondertiteld door de Amara.org gemeenschap",
    " Ondertiteling door de Amara.org gemeenschap"
  ],
  "de": [
    " Untertitelung aufgrund der Amara.org-Community"
    " Untertitel im Auftrag des ZDF für funk, 2017",
    " Untertitel von Stephanie Geiges",
    " Untertitel der Amara.org-Community",
    " Untertitel im Auftrag des ZDF, 2017",
    " Untertitel im Auftrag des ZDF, 2020",
    " Untertitel im Auftrag des ZDF, 2018",
    " Untertitel im Auftrag des ZDF, 2021",
    " Untertitelung im Auftrag des ZDF, 2021",
    " Copyright WDR 2021",
    " Copyright WDR 2020",
    " Copyright WDR 2019",
    " SWR 2021",
    " SWR 2020",
  ],
  "fr": [
    " Sous-titres réalisés para la communauté d'Amara.org",
    " Sous-titres réalisés par la communauté d'Amara.org",
    " Sous-titres fait par Sous-titres par Amara.org",
    " Sous-titres réalisés par les SousTitres d'Amara.org",
    " Sous-titres par Amara.org",
    " Sous-titres par la communauté d'Amara.org",
    " Sous-titres réalisés pour la communauté d'Amara.org",
    " Sous-titres réalisés par la communauté de l'Amara.org",
    " Sous-Titres faits par la communauté d'Amara.org",
    " Sous-titres par l'Amara.org",
    " Sous-titres fait par la communauté d'Amara.org"
    " Sous-titrage ST' 501",
    " Sous-titrage ST'501",
    " Cliquez-vous sur les sous-titres et abonnez-vous à la chaîne d'Amara.org",
    " ❤️ par SousTitreur.com",
  ],
  "it": [
    " Sottotitoli creati dalla comunità Amara.org",
    " Sottotitoli di Sottotitoli di Amara.org",
    " Sottotitoli e revisione al canale di Amara.org",
    " Sottotitoli e revisione a cura di Amara.org",
    " Sottotitoli e revisione a cura di QTSS",
    " Sottotitoli e revisione a cura di QTSS.",
    " Sottotitoli a cura di QTSS",
  ],
  "es": [
    " Subtítulos realizados por la comunidad de Amara.org",
    " Subtitulado por la comunidad de Amara.org",
    " Subtítulos por la comunidad de Amara.org",
    " Subtítulos creados por la comunidad de Amara.org",
    " Subtítulos en español de Amara.org",
    " Subtítulos hechos por la comunidad de Amara.org",
    " Subtitulos por la comunidad de Amara.org"
    " Más información www.alimmenta.com",
    " www.mooji.org",
  ],
  "gl": [
    " Subtítulos realizados por la comunidad de Amara.org"
  ],
  "pt": [
    " Legendas pela comunidade Amara.org",
    " Legendas pela comunidade de Amara.org",
    " Legendas pela comunidade do Amara.org",
    " Legendas pela comunidade das Amara.org",
    " Transcrição e Legendas pela comunidade de Amara.org"
  ],
  "la": [
    " Sottotitoli creati dalla comunità Amara.org",
    " Sous-titres réalisés para la communauté d'Amara.org"
  ],
  "ln": [
    " Sous-titres réalisés para la communauté d'Amara.org"
  ],
  "pl": [
    " Napisy stworzone przez społeczność Amara.org",
    " Napisy wykonane przez społeczność Amara.org",
    " Zdjęcia i napisy stworzone przez społeczność Amara.org",
    " napisy stworzone przez społeczność Amara.org",
    " Tłumaczenie i napisy stworzone przez społeczność Amara.org",
    " Napisy stworzone przez społeczności Amara.org",
    " Tłumaczenie stworzone przez społeczność Amara.org",
    " Napisy robione przez społeczność Amara.org"
    " www.multi-moto.eu",
  ],
  "ru": [
    " Редактор субтитров А.Синецкая Корректор А.Егорова"
  ],
  "tr": [
    " Yorumlarınızıza abone olmayı unutmayın.",
  ],
  "su": [
    " Sottotitoli creati dalla comunità Amara.org"
  ],
  "zh": [
    "字幕由Amara.org社区提供",
    "小編字幕由Amara.org社區提供"
  ]
}

````

**Reply by ILG2021** [reply]
Does this solved?

**Reply by Anto5040** [reply]
I also have in "es": `"es": ["www.alimmenta.com","¡Gracias por ver el vídeo!"]`

**Reply by mayeaux** [reply]
This is a common one for `ja`

```
  "ja": [
    "by H.",
  ],
```

I don't think there's much else of an approach than having people manually reporting them and building up a DB.

**Reply by ILG2021** [reply]
Can not change the code to fix it? it is very likely a fallback or low Confidence result.

**Reply by ryanheise** [reply]
Encountered this for English:

```
<u>Transcribed</u> by https://otter.ai
```

**Reply by ExtReMLapin** [reply]
> Encountered this for English:
> 
> ```
> <u>Transcribed</u> by https://otter.ai
> ```

If they used an AI tool from another company to build their own dataset that's really funny

**Reply by old9** [reply]
encountered this for zh at the beginning silence:
`請不吝點贊訂閱轉發打賞支持明鏡與點點欄目`

**Reply by HeroBlackInk** [reply]
> encountered this for zh at the beginning silence: `請不吝點贊訂閱轉發打賞支持明鏡與點點欄目`

I encounted this problem many times.  i have integrated whisper into the smart speaker

**Reply by SilverGreen93** [reply]
So, doesn't this mean that you need to remove these entries entirely? They confuse out the silence parts of the transcription.
How can I remove these myself? How to rebuild the database?

**Reply by Dadangdut33** [reply]
Using your list and the idea from this discussion i did some testing and mapped out some but not all on each of the language of whisper and [got this result](https://github.com/Dadangdut33/Speech-Translate/blob/master/speech_translate/assets/base_hallucination_filter.json). I tested it on filtering results from live recording and it seems to work even if it seems hacky

**Reply by AlfonsoFR2020** [reply]
I mean, it would be useful in that it is a filter, and it is great work. However, it is also a flawed solution precisely because it's a filter. Is it guaranteed to catch every instance and misbehaviour? How can we know if it does? Or if later on, more entries will keep popping up? It does not address the underlying problem, but only some of its symptoms, like catching a yellow fever and using cold water to lower body temp instead of applying a true cure and proper prophylaxis. Of course it's better than nothing and your contribution is appreciated, but the issue cannot be considered to be solved by this filter alone.

**Reply by Dadangdut33** [reply]
> I mean, it would be useful in that it is a filter, and it is great work. However, it is also a flawed solution precisely because it's a filter. Is it guaranteed to catch every instance and misbehaviour? How can we know if it does? Or if later on, more entries will keep popping up? It does not address the underlying problem, but only some of its symptoms, like catching a yellow fever and using cold water to lower body temp instead of applying a true cure and proper prophylaxis. Of course it's better than nothing and your contribution is appreciated, but the issue cannot be considered to be solved by this filter alone.

Yeah exactly that's why i said it's a hacky solution, not really recommended really

**Reply by lrodeck** [reply]
I have been encountering another one with whisper for german: "Vertraue und glaube, es hilft, es heilt die göttliche Kraft!"

**Comment by vertexgamer** [comment]
It happend to me too, strange enough only with the large-v2 model

**Reply by ILG2021** [reply]
Do you really tried many times and sure of that?

**Reply by old9** [reply]
Perhaps large-v2 is trained on a vast amount of YouTube videos

**Reply by ILG2021** [reply]
so should we change to large-v1?

**Reply by Erick2317** [reply]
I also noticed it when using the large-v2 model, with the medium model this doesn't happen.

**Reply by ILG2021** [reply]
I have find the solution:
https://github.com/EtienneAb3d/WhisperHallu

**Reply by ExtReMLapin** [reply]
None of the solutions cited before will EVER work on realistic audios, like radio coms or anything like that, whisper is trained with a crazy amount of data that no other solution has.

For example, pyannote can diarize yes, but it turns ooga booga as soon as you leaves the "zoom meeting" context it was trained on.

**Reply by xiaoyin199** [reply]
我发现问题是这样的，我使用cli  whisper 没有问题，我在python中使用whisper就出现了问题；
识别最后出现了 Amara.org
识别中间出现了伦桑翻唱】Lun Sang 九月的爱
这就非常奇怪了，而且在python使用whisper识别的准确度和不好！

**Comment by hmijail** [comment]
A funny one: when using `--model large --language ht`, I get `Спасибо` in the middle of the transcription (which is Russian for "thank you" and does not appear in the audio).

The funny thing is that if I input `Спасибо` in Google Translate as Haitian Creole, it **does** translate it to "thank you". But not when, say, French or English are selected.

So I guess both Whisper and Google Translate are using the same bad sources.

**Reply by bunthut** [reply]
I get this in German when using the chatgpt app with integrated speech recognition.

Also the "Untertitel der Amara.org-Community" (German for subtitles of...) and other hallucinations (?)

**Comment by misutoneko** [comment]
I got rid of these with --suppress_tokens "".
I'm actually using whisper-timestamped, but it seems this problem is present in this original version too.

See details here:
https://github.com/linto-ai/whisper-timestamped/issues/105

Continued in https://github.com/openai/whisper/discussions/1488

**Comment by ivansabik** [comment]
I just ended up ignoring rows with a high `no_speech_prob` (> 90%)

**Reply by dgoryeo** [reply]
@ivansabik , do you mean you set `no_speech_threshold` to `0.9` ?

**Reply by ivansabik** [reply]
Actually I was manually excluding those after transcribing since I need to do other post-processing as well. I like keeping all results in case I need to debug later on, but also did not know about this parameter `no_speech_threshold`. It should have the same effect

**Reply by dgoryeo** [reply]
@ivansabik , doe sthis mean you're able to get for each line the `no_speech_prob` in the jason file?

**Reply by ivansabik** [reply]
Yes I get it as part of the segments of the response: [Example]( https://github.com/ivansabik/chairum-corpus/blob/f79e11c70398c698e40927ac3a35c833b57d2806/manual_transcriptions/-8lV1fYCsYA.json#L3-L24):

```javascript
    "segments": [
        {
            "id": 0,
            "seek": 9000,
            "start": 90.0,
            "end": 96.08,
            "text": " \u00bfC\u00f3mo est\u00e1, presidente?",
            "tokens": [
                50364,
                3841,
                28342,
                3192,
                11,
                23852,
                30,
                50668
            ],
            "temperature": 0.0,
            "avg_logprob": -0.7702544668446416,
            "compression_ratio": 1.0673076923076923,
            "no_speech_prob": 0.2541733682155609
        }
```

**Reply by dgoryeo** [reply]
Thanks @ivansabik .

**Reply by rodrigoGA** [reply]
thanks ++

**Reply by dgoryeo** [reply]
I had interpreted from an early comment by @jongwook that there is a corellation between `temperature` , `avg_logprob`, `compression_ratio`, and `no_speech_prob` and haucination.  I'm not an ML engineer, so I refuged to GPT4 to suggest a code to remove hallucination and repetition. This is what GPT4 suggested, and it did work. It removed the hallucination and reptition for me, but it also resulted that many parts of the audio did not get transcribed. 

I'm not sure what is the best way to fine tune the parameters to get a good balance: 

```
 
for r in result["segments"]:
    # Skip audio timestamped after the chunk has ended + offset
    if r["start"] > u[i][-1]["chunk_end"] +10:
        print(f"Skipping segment {r['id']} due to timestamp after chunk end. Text: {r['text']}")
        continue

    if r["avg_logprob"] < -1.0 or r["no_speech_prob"] > 0.7:
        print(f"Skipping segment {r['id']} due to low average log probability or high no speech probability. Text: {r['text']}, avg_logprob: {r['avg_logprob']}, no_speech_prob: {r['no_speech_prob']}")
        continue

    # Check for hallucinations or missed transcriptions
    if r["compression_ratio"] > 2.2 or r["compression_ratio"] < 0.6:
        print(f"Skipping segment {r['id']} due to suspicious compression ratio. Text: {r['text']}, compression_ratio: {r['compression_ratio']}")
        continue

    # Check for uncertain transcriptions
    if r["temperature"] > 0.5:
        print(f"Skipping segment {r['id']} due to high temperature. Text: {r['text']}, temperature: {r['temperature']}")
        continue

    # Check for repetition within the same segment
    if has_repeated_ngrams(get_ngrams(r["text"])):
        print(f"Skipping segment {r['id']} due to detected repetition within the segment. Text: {r['text']}")
        continue


```

**Comment by vivalamovie** [comment]
This is just additional info that this also happens with the ChatGPT Voice feature. I got the string "Untertitel im Auftrag des ZDF für funk, 2017" (which translates to "Subtitles commissioned by ZDF for funk, 2017", funk is the content network of the public broadcasters ARD and ZDF) three times since yesterday. Every time this happened, it was silent in the room, and Whisper seemed to think it recognized something and sent this prompt to the chat.

**Reply by bunthut** [reply]
For me it happens with noise too

**Comment by ExtReMLapin** [comment]
Has anyone tested it with new V3 large model, do you think they filtered their dataset to fix this issue ?

**Reply by BluBb-mADe** [reply]
I am encountering exactly the same silence-hallucination issues with V3. If anything they only expanded the dataset but didn't remove bad data. But honestly, I think it is entirely possible that these hallucinations are just an artifact of the training setup and might even occur with perfectly clean data. I find it suspicious though that in some cases silence results in strings like "www.northstarit.co.uk" from whisper-large-v3 which turns out to be a real website. Maybe it is fashionable nowadays to intentionally pollute training data with things like that for free exposure or something.

**Reply by ExtReMLapin** [reply]
Thanks for the update.
They released it for free, not going to complain about it.

**Reply by vivalamovie** [reply]
> I am encountering exactly the same silence-hallucination issues with V3. If anything they only expanded the dataset but didn't remove bad data. But honestly, I think it is entirely possible that these hallucinations are just an artifact of the training setup and might even occur with perfectly clean data. I find it suspicious though that in some cases silence results in strings like "www.northstarit.co.uk" from whisper-large-v3 which turns out to be a real website. Maybe it is fashionable nowadays to intentionally pollute training data with things like that for free exposure or something.

My guess is because these subtitle notes tend to appear during moments of silence. If you train the system with sufficient data that correlates silence with these notes, the likelihood of them appearing during quiet moments increases significantly.

**Reply by Fusseldieb** [reply]
That's probably correct.

**Reply by SilverGreen93** [reply]
I expected them to appear ONLY in moments of silence, but take a look at my post below. They appear randomly even in moments with speech. I don't understand why...

**Reply by Fusseldieb** [reply]
Maybe your audio is too quiet. Try adding a compressor.

**Comment by SilverGreen93** [comment]
Hi,
I randomly get the same message in Hungarian in kdenlive using Hungarian Large model.
What's the issue exactly? Audio is not silence here, has a lot of speech.

![image](https://github.com/openai/whisper/assets/24845602/89de94e2-36a6-4ea1-b920-eeb717d748c6)

Thanks!

**Reply by polaroi8d** [reply]
@SilverGreen93 hahahahahhha we got the same 😄  Did you manage to filter out these texts?

**Comment by AlfonsoFR2020** [comment]
Yes, I have found this when using the app to talk with ChatGPT on Android (the one with 10M downloads as of today). I believe the problem is more of a managerial than of a programming nature. It seems that conversations are being rerouted via the API to an intermediary who performs the translations (sometimes really badly, so I tend to push-to-talk or either to type, but not to use the continual speech feature identified by the "headphones" icon, which produces even worse dictations/translations, especially for Castillian/European Spanish). Sometimes, during these audio conversations, I can hear clicks and beeps in the background when they should not be heard, so it really seems to me that they are sending my data straight through a clickfarm in an offshore call center, like in the last two episodes of "Silicon Valley -3rd season". These "features" are temperamental and sometimes will stop responding altogether and cut your conversation off. Some other times they will alter the contents of your texts randomly (e.g.: from "morning" to "afternoon"). I am really worried about the possibility of eavesdropping, IP theft, data security, and, of course, performance and effectiveness. Also, it is really annoying that the tag about "Amara.org" is being inserted randomly within conversations: it would seem that this company wants to make themselves heard as if they were claiming credit for some of the features, I do not know if justifiedly or not, but certainly in a technically terrible and obnoxious manner. IF you have to choose, as said, the push-to-talk feature works best regarding transcriptional accuracy.

**Reply by bunthut** [reply]
Lol. You don't really think that a company developing that powerful language models, will have people transcribing speech to text and sending it back? That's hilarious.

**Reply by Fusseldieb** [reply]
Confidentially incorrect. You really think someone can transcribe a whole 10 minute audio in 10 seconds? Technology is moving pretty fast, and if you believe this what you said is real, it's best for you to take a minute off the internet.

**Comment by AlfonsoFR2020** [comment]
Now after a push to talk it sometimes spits out stuff about a "2020 Copyright" instead of transcribing... really weird.

**Comment by dgoryeo** [comment]
Am I write to think that we should collect the audio segments that produce those hallucination lines, and then fine tune (train) Whisper model with those audio segments? Has anyone already experimented with that?

**Comment by fabswt** [comment]
it makes sense this would happen. the model was obviously trained on subtitles and those specific subs typically show at the end of a video when there's nothing being said

i can think of two ways to fix this:

1. reject empty/near-empty audio
2. filter for such messages like @doublex offered

i have 2.5K transcriptions with "Amara" in it (on [gliglish.com](https://gliglish.com)), will test things on my end and report results.

**Reply by ryanheise** [reply]
You can detect these Amara hallucinations (and others) with #1838 or plugging in a dedicated VAD.

**Reply by EtienneAb3d** [reply]
@fabswt 
Several ideas like these were tested her (on some others are in preparation and will come very soon, perhaps today):
https://github.com/EtienneAb3d/WhisperHallu
;-)

**Comment by 6DiegoDiego9** [comment]
Any news about this issue?
It's 1 year and 4 months after the original post, and it's still impossible to use ChatGPT voice conversation, especially outdoors, because of these obsessive Amara.org hallucinations.

**Reply by leye0** [reply]
Yeah. Programmatically, I can filter out the "subtitles" results but these useless calls to Whisper speech-to-text are costly.

**Comment by deviato** [comment]
I can confirm that the same thing keeps happening in Italian with sentences like “Sottotitoli e revisione a cura di QTSS” (Subtitles and revision by QTSS).

The models were clearly trained with audio from pirated movies and subtitles taken from the open community without prior filtering of credits and/or descriptive notes. All the models are garbage, the training should be repeated from start, with consistent data.

**Reply by misutoneko** [reply]
The fact that these translations are different in different languages can actually be used as a way to detect problematic parts of the audio in post-processing. It requires the suspicious parts to be processed twice though.

**Reply by dgoryeo** [reply]
Interesting point. Would you elaborate please about the post-processing approach.  Thanks.

**Reply by misutoneko** [reply]
Here's the old thread from 2023 that describes my (then-current) process for english audio:
https://github.com/linto-ai/whisper-timestamped/issues/82#issuecomment-1511092786

Yeah pretty crude, I don't know how you would implement that in Python.
Set the language separately for each 30-sec portion and see which tokens come up, something like that, I guess?
And then if you detect any of the "suspicious" tokens, do a re-run with a different language.

**Reply by leye0** [reply]
Omg. I was using the vocal version of ChatGPT with a loud white noise in background and it was continuously sending these false positives to the assistant ![image](https://github.com/user-attachments/assets/c8a64016-bc08-4852-8311-718e4325a3e8)

So the vocal version of ChatGPT is still using that same model with the "subtitles" bug.

**Comment by quinncomendant** [comment]
I just got this “easter egg” when making a transcription of Spanish:

<img width="529" alt="image" src="https://github.com/user-attachments/assets/c0a1025b-4da8-4c81-9131-e0d92de661f8" />

This is from the Aiko app, which uses Whisper.

**Comment by leye0** [comment]
> I can confirm that the same thing keeps happening in Italian with sentences like “Sottotitoli e revisione a cura di QTSS” (Subtitles and revision by QTSS).
> 
> 
> 
> The models were clearly trained with audio from pirated movies and subtitles taken from the open community without prior filtering of credits and/or descriptive notes. All the models are garbage, the training should be repeated from start, with consistent data.

Yeah, that's very unprofessional for a commercial product. It's weird that their engineers didn't think about removing this credit.

It also means that in some specific silent moments, we could obtain scene descriptions like "Ominous music playing" 😆

**Comment by ZigaoWang** [comment]
![CleanShot 2025-03-31 at 14 57 37@2x](https://github.com/user-attachments/assets/9109d410-8a75-4979-a66f-92b34561f104)
There is also bias for MING PAO CANADA | MING PAO TORONTO.

**Comment by NielsMayer** [comment]
Another interesting source of hallucination (see https://github.com/openai/whisper/discussions/1873#discussioncomment-8613493 )   are sequences of "CCoSp4 3.50"

These come from Figure Skating competition tables :-)

https://ijs.usfigureskating.org/leaderboard/results/2024/33923/SEGM024.html

See also https://github.com/openai/whisper/discussions/1873#discussioncomment-10650388 : "Suggestion: don't use train your language models on Ice Skating videos, or classical music will trigger transcription of figure skating notations. :-)"

**Comment by drAlesya** [comment]
Buongiorno

**Comment by leye0** [comment]
It still surprises me that to this day, there has been no scandal about OpenAI training Whisper on torrents.

**Reply by vertexgamer** [reply]
As long as the model is free i don't see the problem.
If i illegally scrape CT scans and release a model to the public that can spot tumors it's ok, if i put it behind a paywall and don't distribute it it's not ok

**Reply by deviato** [reply]
> As long as the model is free i don't see the problem. If i illegally scrape CT scans and release a model to the public that can spot tumors it's ok, if i put it behind a paywall and don't distribute it it's not ok

Apart from the fact that copyright doesn't work that way. And anyway, OpenAI sells its subscription products with these features.

**Reply by vertexgamer** [reply]
I'm talking from a morality POV not a legal one, only hypocrites would get pissed off about copyright.
Still the subscription is not necessary for using whisper, so it's ok
